docs/pipelines.md

```diff
@@ -6,157 +6,111 @@ To accomplish this, the EKS SSP - Reference Solution leverages the [`Pipelines`]
-
-export class ClusterStage extends cdk.Stage {
-    constructor(scope: cdk.Stack, id: string, props?: cdk.StageProps) {
-        super(scope, id, props);
-
-        // Setup platform team
-        const accountID = props?.env?.account
-        const platformTeam = new team.TeamPlatform(accountID!)
-        const teams: Array<ssp.Team> = [platformTeam];
-
-        // Add-ons for the cluster.
-        const addOns: Array<ssp.ClusterAddOn> = [
-            new ssp.NginxAddOn,
-            new ssp.ArgoCDAddOn,
-            new ssp.CalicoAddOn,
-            new ssp.MetricsServerAddOn,
-            new ssp.ClusterAutoScalerAddOn,
-            new ssp.ContainerInsightsAddOn,
-        ];
-        new ssp.EksBlueprint(this, { id: 'blueprint-cluster', addOns, teams }, props);
-    }
-}
+const blueprint = ssp.EksBlueprint.builder()
+    ...; // configure your blueprint builder
+
+ ssp.CodePipelineStack.builder()
+    .name("ssp-eks-pipeline")
+    .owner("aws-samples")
+    .repository({
+        repoUrl: 'ssp-eks-patterns',
+        credentialsSecretName: 'github-token',
+        branch: 'main'
+    })
 ```
 
-## Adding stages to the pipeline. 
-
-Once a stage is created, we simply add it to the pipeline. 
+Note: the above code depends on the AWS secret `github-token` defined in the target account/region. The secret may be fined in one main region, and replicated to all target regions. 
 
-```typescript
-const dev = new ClusterStage(this, 'blueprint-stage-dev')
-pipeline.addApplicationStage(dev);
-```
+## Creating stages 
 
-Adding stages to deploy multiple pipelines is trivial. 
+Once our pipeline is created, we need to define `stages` for the pipeline. To do so, we can leverage `ssp.StackStage` convenience class and builder support for it. Let's continue leveraging the pipeline builder defined in the previous step.  
 
 ```typescript
-const dev = new ClusterStage(this, 'blueprint-stage-dev')
-pipeline.addApplicationStage(dev);
-
-const test = new ClusterStage(this, 'blueprint-stage-test')
-pipeline.addApplicationStage(test);
+const blueprint = ssp.EksBlueprint.builder()
+    ...; // configure your blueprint builder
+
+ssp.CodePipelineStack.builder()
+    .name("ssp-eks-pipeline")
+    .owner("aws-samples")
+    .repository({
+        //  your repo info
+    }) 
+    .stage({
+        id: 'us-west-1-managed-ssp-test',
+        stackBuilder: blueprint.clone('us-west-1') // clone the blueprint to customize for the stage. You can add more add-ons, teams here. 
+    })
+    .stage({
+        id: 'us-east-2-managed-ssp-prod',
+        stackBuilder: blueprint.clone('us-east-2'), // clone the blueprint to customize for the stage. You can add more add-ons, team, here.
+        stageProps: {
+            manualApprovals: true
+        }
+    })
 ```
 
-We can also add manual approvals for production stages. 
+Consider adding `ArgoCDAddOn` with specific workload bootstrap repository to automatically bootstrap workloads in the provisioned clusters.
```

Comment:
Grammar change - Consider adding `ArgoCDAddOn` with your specific workload repository to automatically bootstrap those workloads in the provisioned clusters.

Reply:
addressed